lnls-sirius/dev-packages
# Allow removal of correctors in PSSOFB mode

# Changes made:
 - PS -> Improve interpolation in excdata:
    - Only call linear_extrapolation when needed.
   - When xvals is NaN interp will be NaN: before this change the code wasn't able to separate NaN inputs from left extrapolation.
- PSSOFB -> Only send signal to IOC to update variables when PV is connected.
- PSBSMP -> Handle NaNs in setpoints properly:
   - If values to be set has NaNs, filter them;
   - If all entries of values are NaNs, do not send setpoints and return ok.
- SOFB -> When corrector is not being used in PSSOFB mode, try not to send setpoints to it.
- SOFB -> Do not check status or configure correctors and BPMs not used in correction.

# Tests needed before merging this PR:

 - Launch IOC without errros;
 - Configure BPM and correctors parameters and check if parameters of correctors and BPMs not used were not changed;
 - Change the some parameter of BPM and corrector not in loop and check if status is still ok;
 - Close the loop with all BPMs and Correctors selected in PSSOFB mode;
 - Open the loop, remove a corrector and close the loop again;
 - Remove a corrector without opening the loop;
 - Disconnect some UDC from a beaglebone and try to close the loop;
 - Disconnect some Beaglebone from ethernet and try to close the loop;
